cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -457,7 +464,11 @@ private static void extractBala(Path balaFilePath, Path balaFileDestPath, String
 
         try (FileSystem zipFileSystem = FileSystems.newFileSystem(zipURI, new HashMap<>())) {
             Path packageRoot = zipFileSystem.getPath("/");
-            List<Path> paths = Files.walk(packageRoot).filter(path -> path != packageRoot).toList();
+            List<Path> paths;
+            try (Stream<Path> pathStream = Files.walk(packageRoot)) {
```

Comment:
This can be merged with the outer try block. I will send those changes in a sepaerta PR